lab_5_scrapper/scrapper.py

```diff
@@ -1,82 +1,199 @@
 """
 Crawler implementation
 """
+
+from pathlib import Path
 from typing import Pattern, Union
 
+import re
+import json
+from bs4 import BeautifulSoup
+import requests
+from requests import HTTPError
+
+from core_utils.config_dto import ConfigDTO
+from core_utils.constants import ASSETS_PATH, CRAWLER_CONFIG_PATH
+
+
+class IncorrectSeedURLError(Exception):
+    pass
+
+
+class NumberOfArticlesOutOfRangeError(Exception):
+    pass
+
+
+class IncorrectNumberOfArticlesError(Exception):
+    pass
+
+
+class IncorrectHeadersError(Exception):
+    pass
+
+
+class IncorrectEncodingError(Exception):
+    pass
+
+
+class IncorrectTimeoutError(Exception):
+    pass
+
+
+class IncorrectVerifyError(Exception):
+    pass
+
 
 class Config:
     """
     Unpacks and validates configurations
     """
 
+    seed_urls: list[str]
+    total_articles_to_find_and_parse: int
+    headers: dict[str, str]
+    encoding: str
+    timeout: int
+    verify_certificate: bool
+    headless_mode: bool
+
     def __init__(self, path_to_config: Path) -> None:
         """
         Initializes an instance of the Config class
         """
-        pass
+        self.path_to_config = path_to_config
+        self._validate_config_content()
+        self._extract_config_content()
 
     def _extract_config_content(self) -> ConfigDTO:
         """
         Returns config values
         """
-        pass
+        with open(self.path_to_config, 'r', encoding='utf-8') as f:
+            config_params = json.load(f)
+
+            config_dto = ConfigDTO(seed_urls=config_params['seed_urls'],
```

Comment:
it is ok, but try to minimize the code with unpacking of dictionaries with asterix, try, you will enjoy